library/ssl_tls.c

```diff
@@ -6151,13 +6156,46 @@ static int ssl_preset_default_hashes[] = {
 #if defined(MBEDTLS_SHA256_C)
     MBEDTLS_MD_SHA256,
 #endif
-#if defined(MBEDTLS_SHA224_C)
-    MBEDTLS_MD_SHA224,
+    MBEDTLS_MD_NONE
+};
+#endif
+
+#if defined(MBEDTLS_ECP_C)
+/* The selection should be the same as mbedtls_x509_crt_profile_default in
+ * x509_crt.c, plus Montgomery curves for ECDHE. Here, the order matters:
+ * curves with a lower resource usage come first.
+ * See the documentation of mbedtls_ssl_conf_curves() for what we promise
+ * about this list.
+ */
+static mbedtls_ecp_group_id ssl_preset_default_curves[] = {
+#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED)
+    MBEDTLS_ECP_DP_CURVE25519,
 #endif
-#if defined(MBEDTLS_SHA1_C) && defined(MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_KEY_EXCHANGE)
-    MBEDTLS_MD_SHA1,
+#if defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED)
+    MBEDTLS_ECP_DP_SECP256R1,
 #endif
-    MBEDTLS_MD_NONE
+#if defined(MBEDTLS_ECP_DP_SECP256K1_ENABLED)
+    MBEDTLS_ECP_DP_SECP256K1,
```

Comment:
As was just reported in https://github.com/ARMmbed/mbedtls/issues/4613, [RFC 8422 §5.1.1](https://datatracker.ietf.org/doc/html/rfc8422#section-5.1.1) deprecates secp256k1. Should we offer it at all? Only as a last resort?

Reply:
Nope, let's remove it from the default. I'm not sure anybody ever used it for TLS anyway. (We added it to Mbed TLS because it's used in bitcoin, fwiw.)

Reply:
Argument for keeping it: it's allowed in X.509. Or should we disable it by default in X.509? It doesn't seem to be deprecated there.

Reply:
Honestly, I highly doubt anybody's using secp256k1 in certificates, so it probably doesn't matter in practice. But as a matter of principle, it's indeed a bit harder to justify excluding it from the X.509 profile.